pkg/network/ebpf/c/protocols/events.h

```diff
@@ -0,0 +1,91 @@
+        u32 zero = 0;                                                                             \
+        batch_state_t *batch_state =  bpf_map_lookup_elem(&name##_batch_state, &zero);            \
+        if (batch_state == NULL) {                                                                \
+            return;                                                                               \
+        }                                                                                         \
+                                                                                                  \
+        batch_key_t key = get_batch_key(batch_state->idx);                                        \
+        batch_data_t *batch = bpf_map_lookup_elem(&name##_batches, &key);                         \
+        if (batch == NULL) {                                                                      \
+            return;                                                                               \
+        }                                                                                         \
+                                                                                                  \
+        if (name##_batch_full(batch)) {                                                           \
+            log_debug(STR(name) " enqueue error: dropping element because batch is full. "        \
+                      "cpu=%d batch_idx=%d\n",                                                    \
+                      bpf_get_smp_processor_id(), batch->idx);                                    \
+            return;                                                                               \
+        }                                                                                         \
+                                                                                                  \
+        u32 offset = batch->len*sizeof(value);                                                    \
+        if (offset < 0 || offset+sizeof(value)>sizeof(batch->data)) {                             \
+            return;                                                                               \
+        }                                                                                         \
+                                                                                                  \
+        bpf_memcpy(&batch->data[offset], element, sizeof(value));                                 \
+        /* cap and obj_size are used as metadata by userspace only */                             \
+        batch->cap = batch_size;                                                                  \
+        batch->element_size = sizeof(value);                                                      \
+        batch->len++;                                                                             \
```

Comment:
```suggestion
        batch->entries++;                                                                             \
```

Reply:
I thought keeping `len` and `cap` since they're already used in Go so the meaning is hopefully self explanatory